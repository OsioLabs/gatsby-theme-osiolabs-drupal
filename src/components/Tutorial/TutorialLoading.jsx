import React from 'react';
import PropTypes from 'prop-types';
import { Container, Placeholder } from 'semantic-ui-react';
import TutorialTeaser from './TutorialTeaser';

const TutorialLoading = props => (
  <Container>
    <TutorialTeaser {...props} />
    <p />
    <Placeholder fluid>
      <Placeholder.Header>
        <Placeholder.Line />
      </Placeholder.Header>
      <Placeholder.Paragraph>
        <Placeholder.Line />
        <Placeholder.Line />
        <Placeholder.Line />
        <Placeholder.Line />
        <Placeholder.Line />
      </Placeholder.Paragraph>
      <Placeholder.Header>
        <Placeholder.Line />
      </Placeholder.Header>
      <Placeholder.Paragraph>
        <Placeholder.Line />
        <Placeholder.Line />
        <Placeholder.Line />
      </Placeholder.Paragraph>
    </Placeholder>
  </Container>
);

TutorialLoading.propTypes = {
  title: PropTypes.string.isRequired,
  summary: PropTypes.string.isRequired,
};

export default TutorialLoading;
