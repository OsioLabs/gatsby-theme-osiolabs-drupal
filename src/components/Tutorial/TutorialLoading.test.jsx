import React from 'react';
import { render } from '@testing-library/react';

import TutorialLoading from './TutorialLoading';

// We don't need to test this inner component, which requires Oauth context,
// since it has it's own tests.
jest.mock('./TutorialTeaser', () => () => 'TutorialTeaser');

describe('Component: <TutorialLoading />', () => {
  it('renders correctly', () => {
    // Basic snapshot test.
    const { asFragment } = render(
      <TutorialLoading
        title="Tacos Rock"
        summary="<p>Tacos <strong>go here</strong> ...</p>"
      />
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});
