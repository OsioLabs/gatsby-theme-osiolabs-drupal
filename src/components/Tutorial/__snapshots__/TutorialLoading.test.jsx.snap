// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TutorialLoading /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui container"
  >
    TutorialTeaser
    <p />
    <div
      class="ui fluid placeholder"
    >
      <div
        class="header"
      >
        <div
          class="line"
        />
      </div>
      <div
        class="paragraph"
      >
        <div
          class="line"
        />
        <div
          class="line"
        />
        <div
          class="line"
        />
        <div
          class="line"
        />
        <div
          class="line"
        />
      </div>
      <div
        class="header"
      >
        <div
          class="line"
        />
      </div>
      <div
        class="paragraph"
      >
        <div
          class="line"
        />
        <div
          class="line"
        />
        <div
          class="line"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
